agent/agent.go

```diff
@@ -534,7 +534,13 @@ func (a *Agent) reloadWatches(cfg *config.RuntimeConfig) error {
 	// Compile the watches
 	var watchPlans []*watch.Plan
 	for _, params := range cfg.Watches {
-		// Parse the watches, excluding the handler
+		if handlerType, ok := params["handler_type"]; !ok {
+			params["handler_type"] = "script"
+		} else if handlerType != "http" && handlerType != "script" {
+			return fmt.Errorf(fmt.Sprintf("Handler type '%s' not recognized", params["handler_type"]))
```

Comment:
This shouldn't need the unnecessary inner `fmt.Sprintf(..)

Reply:
Fixed